calculator.c

```diff
@@ -366,27 +369,33 @@ Symbol tokenType(token tk)
 		case digit:
 			ret = value;
 			break;
+		default:
+			break;
 	}
 	return ret;
 }
 
-token tokenizeNumber(char *str)
-{
-	;
-}
 
 int tokenize(char *str, char *(**tokensRef))
 {
 	char** tokens = NULL;
+	char** tmp = NULL;
 	char* ptr = str;
 	char ch = '\0';
 	int numTokens = 0;
-	while(ch = *ptr++)
+	char* tmpToken = malloc((MAXTOKENLENGTH+1) * sizeof(char));
+	if (!tmpToken)
+	{
+		fprintf(stderr, "Malloc of temporary buffer failed\n");
+		return 0;
+	}
+	while((ch = *ptr++))
 	{
 		if(type(ch) == invalid) // Stop tokenizing when we encounter an invalid character
 			break;
 
 		token newToken = NULL;
+		memset(tmpToken, '\0', MAXTOKENLENGTH+1);
```

Comment:
How much does zeroing memory impact performance? I'm thinking that since the tokenization process sets all of the bytes through and including the null terminator, it might not need to zero the entire token. Given the `if` on [line 527](https://github.com/btmills/calculator/pull/6/files#diff-0e65cd880e53ce3dd36e6bbae63d6ca5R527), that might still require explicitly zeroing the first byte here.